core/lib/src/outcome.rs

```diff
@@ -484,6 +484,96 @@ impl<S, E, F> Outcome<S, E, F> {
         }
     }
 
+    /// Maps an `Outcome<S, E, F>` to an `Outcome<T, E, F>` by applying the
+    /// function `f` to the value of type `S` in `self` if `self` is an
+    /// `Outcome::Success`.
```

Comment:
I think the shorter wording of [Result::and_then](https://doc.rust-lang.org/nightly/std/result/enum.Result.html#method.and_then) is clearer. For example: "Calls `f` if `self` is a [Success], otherwise returns the [Failure]/[Forward] value of `self`"

The same goes for the other two methods (compare to `Result::or_else`).

Reply:
Done